### PR TITLE
[SD-133] Make disciplina node implementation work with latest `master` of snowdrop

### PR DESCRIPTION
**Pull request by pva701**

### CODE REVIEW

File: witness/src/Dscp/Witness/Logic/Processing.hs [hunk 1]
```diff
@@ -57,17 +58,20 @@ createBlock newSlot = do
...
+    let stateDBM = nsStateDBActions sdActions (SD.RememberForProof True)
 
     (blockCS, stateCS) <- reify blockPrefixes $ \(_ :: Proxy ps) ->
         let actions = SD.constructCompositeActions @ps (SD.dmaAccessActions blockDBM)
                                                         (SD.dmaAccessActions stateDBM)
             rwComp = do
               sblock <- SD.liftERoComp $ expandBlock block
-              SD.applyBlockImpl toVerify blkStateConfig sblock
+              -- getCurrentTime requires MonadIO
+              let osParams = SD.unOSParamsBuilder sdOSParamsBuilder $ UTCTime (toEnum 0) (toEnum 0)
```
Comment (georgeee) [inline witness/src/Dscp/Witness/Logic/Processing.hs]: Looks weird... Maybe we quickly perform refactoring for osparams to integrate them properly

Comment (pva701) [reply witness/src/Dscp/Witness/Logic/Processing.hs]: ahahahah nu nu, quickly.

Comment (pva701) [reply witness/src/Dscp/Witness/Logic/Processing.hs]: it looks as weird as implementation of os params in snowdrop, imho

Comment (pva701) [reply witness/src/Dscp/Witness/Logic/Processing.hs]: >Maybe we quickly perform refactoring for osparams to integrate them properly

it's not so easy, I think. you should somehow manage os params, I mean create them, recalculate in some moments, IMHO, it's not so easy to design them properly. abstract type is the smallest thing in this refactoring.

Comment (georgeee) [reply witness/src/Dscp/Witness/Logic/Processing.hs]: Let's at least pass start time as UtcTime 0 0 so it's clear it's not used anyhow

**Comment by pva701** [comment]
@georgeee done

**Pull request closed without merging**